src/handlers/Registrar.ts

```diff
@@ -1,51 +1,54 @@
-import { type Context, type Event } from "ponder:registry";
+import { type Context } from "ponder:registry";
 import { domains, registrations } from "ponder:schema";
 import { type Hex, namehash } from "viem";
 import { isLabelValid, makeSubnodeNamehash, tokenIdToLabel } from "../lib/ens-helpers";
-import { NsReturnType } from "../lib/plugins";
 import { upsertAccount, upsertRegistration } from "../lib/upserts";
 
-type NsType<T extends string> = NsReturnType<T, "/eth">;
-
 const GRACE_PERIOD_SECONDS = 7776000n; // 90 days in seconds
 
-export const makeRegistryHandlers = (baseName: `.${`${string}`}eth`) => {
-  const baseNameNode = namehash(baseName.slice(1));
+/**
+ * A factory function that returns Ponder indexing handlers for a specified index name/subname.
+ */
+export const makeRegistryHandlers = (indexedSubname: `${string}eth`) => {
+  const indexedSubnameNode = namehash(indexedSubname);
 
   async function setNamePreimage(context: Context, name: string, label: Hex, cost: bigint) {
     if (!isLabelValid(name)) return;
 
-    const node = makeSubnodeNamehash(baseNameNode, label);
+    const node = makeSubnodeNamehash(indexedSubnameNode, label);
     const domain = await context.db.find(domains, { id: node });
     if (!domain) throw new Error("domain expected");
 
     if (domain.labelName !== name) {
       await context.db
         .update(domains, { id: node })
-        .set({ labelName: name, name: `${name}${baseName}` });
+        .set({ labelName: name, name: `${name}${indexedSubname}` });
     }
 
     await context.db.update(registrations, { id: label }).set({ labelName: name, cost });
   }
 
   return {
-    get baseNameNode() {
-      return baseNameNode;
+    get indexedSubnameNode() {
+      return indexedSubnameNode;
     },
 
     async handleNameRegistered({
       context,
       event,
     }: {
       context: Context;
-      event: Omit<Event<NsType<"BaseRegistrar:NameRegistered">>, "name">;
+      event: {
+        block: { timestamp: bigint };
```

Comment:
i agree on splitting out the args for clarity, but can you type the block argument as a ponder `Block`?